### ISSUE TITLE
Android (GrapheneOS) - no internet after installation

### ISSUE THREAD

**Issue by b-ryan** [tailscale/tailscale#6516]
### What is the issue?

Hello!

I have just installed GrapheneOS on a Pixel 6a, I haven't made any sort of customizations on it to the networking. I installed Tailscale and after Tailscale sets itself up as the always-on VPN, the internet doesn't work at all. It appears as though DNS is failing. The error I see in my browser is

> [site]'s DNS address could not be found. Diagnosing the problem.
>
> DNS_PROBE_POSSIBLE

I have attempted to disable all custom DNS settings, but to no avail. I'm not sure how to debug this. I installed a terminal emulator, but the tools I'm used to don't seem to be there (telnet, dig, nslookup, curl, wget). 

### Steps to reproduce

? not sure other than installing GrapheneOS and Tailscale together. 

### Are there any recent changes that introduced the issue?

Brand new installation of GrapheneOS. Only installed a handful of apps that seem unlikely to be related. 

### OS

Android

### OS version

2022111800

### Tailscale version

1.33.97

### Bug report

BUG-1fb35b338f88058a4c5b1bdf6a2df4c92991a3d4be4700ddca6f67cbf28d67de-20221125211124Z-a792424d8669ec32

**Comment by DentonGentry** [comment]
When "Override local DNS" is not set in https://login.tailscale.com/admin/dns, the client needs to read the list of DNS servers out from the underlying platform. Android does not make it easy to do this, we have a few mechanisms to do so which work on different versions of the Android OS.

Earlier in the day we see it successfully retrieve the DNS servers the platform received through DHCP:

```
2022-11-25 20:46:02.844919521 +0000 UTC: getDnsConfigFromLinkProperties: 2600:4040:q:r:s:t:u:v 192.168.1.200 ;localdomain
2022-11-25 20:46:02.845600144 +0000 UTC: getDnsServersFromSystemProperties:
2022-11-25 20:46:02.873690761 +0000 UTC: getDnsServersFromNetworkInfo: 192.168.1.200
...
2022-11-25 20:46:02.876831386 +0000 UTC: 19.0M/248.4M dns: Resolvercfg: {Routes:{.:[2600:4040:q:r:s:t:u:v 192.168.1.200] example.dev.:[192.168.1.200]} Hosts:9 LocalDomains:[]}
2022-11-25 20:46:02.876936529 +0000 UTC: 19.0M/248.4M dns: OScfg: {Nameservers:[100.100.100.100] SearchDomains:[example.dev. localdomain.] MatchDomains:[] Hosts:[]}
```
--------

Shortly before the bugreport it looks like the app was disconnected, connected again, disconnected, and then the bug report generated. We don't see it even trying to retrieve the platform's DNS servers, it just receives an empty DNS config.

```
2022-11-25 20:50:44.554078955 +0000 UTC: 24.7M/254.8M dns: Set: {DefaultResolvers:[] Routes:{} SearchDomains:[] Hosts:6}
2022-11-25 20:50:44.554226293 +0000 UTC: 24.7M/254.8M dns: Resolvercfg: {Routes:{} Hosts:6 LocalDomains:[]}
2022-11-25 20:50:44.554435522 +0000 UTC: 24.7M/254.8M dns: OScfg: {Nameservers:[] SearchDomains:[] MatchDomains:[] Hosts:[]}
...
2022-11-25 20:52:07.804108658 +0000 UTC: 17.1M/254.8M control: setPaused(true)
```

**Comment by dockets-realer** [comment]
Face the same issue with Chrome Cast with Google TV (Android TV 12)

**Comment by DentonGentry** [comment]
@dockets-realer Please file a new issue and include a bugreport (generated from the menu in the Tailscale app on the Chromecast).

**Comment by b-ryan** [comment]
I just connected Tailscale again and this time, in the android VPN settings, I disabled the setting "Block connections without VPN." Now the internet works.

I also noticed that going directly to a Tailscale IP address works whether or not this setting is checked.

This is starting to make some sense, as Tailscale must only be sending traffic meant for other Tailscale machines through the Tailnet, otherwise using Android's networking and this is being blocked by that setting.

I don't _think_ I turned this setting on. I have repeatedly told Android to "Forget this VPN" while I have been debugging and each time Tailscale creates the VPN, this setting has been enabled. Should it be disabled by default?

**Comment by DentonGentry** [comment]
Tailscale's default is to be a splitVPN, enabling connectivity to Tailscale IP addresses while letting all other accesses go over the public Internet. The Tailscale app definitely does not set "Block connections without VPN."

**Comment by codethief** [comment]
I just ran into exactly the same issue on GrapheneOS. Thank you all for figuring this out! :)

**Comment by codethief** [comment]
Unfortunately, I need to bring this up again: I disabled "Block connections without VPN" but left "Always-on VPN" enabled. Now, when I was on wifi everything worked fine (see my previous comment). But I now noticed that whenever I leave my home network and my phone is on 5G, my other apps on my phone stop having internet access. Meanwhile, Tailscale works fine (though sometimes it fails to connect, too). On the other hand, as soon as I disable Tailscale, all my apps have internet again.

Notably, even if I disable "Always-on VPN", the situation is the same: Whenever I'm on 5G, I either have access to the internet, or I am connected to my tailnet, but not both.

How can I debug this? Tailscale Android doesn't seem to have an option to view the logs or anything.

**Comment by thestinger** [comment]
> Tailscale's default is to be a splitVPN, enabling connectivity to Tailscale IP addresses while letting all other accesses go over the public Internet. The Tailscale app definitely does not set "Block connections without VPN."
> 
> I think this means the issue is caused by a policy decision by GrapheneOS to default all VPNs to block connections not using the VPN. To be honest that is a reasonable choice if your overwhelming focus is on privacy, but it does break splitVPNs like Tailscale.
> 
> The commit which changed the default is: [GrapheneOS/platform_frameworks_base@0bd1dbf](https://github.com/GrapheneOS/platform_frameworks_base/commit/0bd1dbf60fe641dbdd7f60b77ace71c3328c7e37)

We enable it by default, but the option is still there on the stock OS and will be broken in the same way. It doesn't work differently on Android 13 and GrapheneOS beyond us changing the default and fixing some issues with backports.

**Comment by thestinger** [comment]
The standard split tunneling is known to be broken with the block connections feature. I think it's possible to do routing in a different way where it won't be broken, by passing everything through your app.
